apps/root/src/pages/position-detail/components/graph-container/components/swap-price-graph/index.tsx

```diff
@@ -134,35 +87,35 @@ const AveragePriceGraph = ({ position }: AveragePriceGraphProps) => {
           position.chainId,
           swappedActions.length * FREQUENCY_TO_MULTIPLIER[position.swapInterval.toString()],
           FREQUENCY_TO_PERIOD[position.swapInterval.toString()],
-          swappedActions[0].tx.timestamp.toString()
+          swappedActions[swappedActions.length - 1] && swappedActions[swappedActions.length - 1].tx.timestamp.toString()
```

Comment:
why this change? was it wrong?

Reply:
[0] is the first timestamp of the position and we needed the last one for this `end` parameter